src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateCopy, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        action.keysToRemove.forEach(key => {
+          delete stateCopy[key];
+        });
+        break;
+      }
+
+      case 'clear': {
+        Object.keys(stateCopy).forEach(key => {
+          delete stateCopy[key];
+        });
```

Comment:
we clear all values from our object so can we just assign new empty object to avoid additional loop?